### PR TITLE
User compset

### LINKED ISSUE #1315: Error trying to create a case with --user-compset
When I try to create a case with `--user-compset` from cime master, I get the following traceback:

```
Traceback (most recent call last):
  File "./create_newcase", line 204, in <module>
    _main_func(__doc__)
  File "./create_newcase", line 187, in _main_func
    run_unsupported=run_unsupported, answer=answer)
  File "/glade/p/work/sacks/cime/scripts/Tools/../../scripts/lib/CIME/case.py", line 745, in configure
    if grid_name in science_support:
TypeError: argument of type 'bool' is not iterable
```

Here is the complete command and output:

```
[yslogin4:/glade/p/work/sacks/cime/scripts]$ ./create_newcase -case test_a_user --user-compset -compset 1850_DATM%NYF_SLND_DICE%SSMI_DOCN%DOM_DROF%NYF_SGLC_SWAV -res f45_g37_rx1 --pesfile ../src/drivers
/mct/cime_config/config_pes.xml
Could not find a compset match for either alias or longname in 1850_DATM%NYF_SLND_DICE%SSMI_DOCN%DOM_DROF%NYF_SGLC_SWAV
Pes setting: grid          is a%4x5_l%null_oi%gx3v7_r%rx1_g%null_w%null_m%gx3v7
Pes setting: compset       is 1850_DATM%NYF_SLND_DICE%SSMI_DOCN%DOM_DROF%NYF_SGLC_SWAV
Pes setting: tasks       is {'NTASKS_ATM': -1, 'NTASKS_ICE': -1, 'NTASKS_CPL': -1, 'NTASKS_LND': -1, 'NTASKS_WAV': -1, 'NTASKS_ROF': -1, 'NTASKS_OCN': -1, 'NTASKS_GLC': -1, 'NTASKS_ESP': -1}
Pes other settings: {'comment': 'none'}
 Compset is: 1850_DATM%NYF_SLND_DICE%SSMI_DOCN%DOM_DROF%NYF_SGLC_SWAV
 Grid is: a%4x5_l%null_oi%gx3v7_r%rx1_g%null_w%null_m%gx3v7
 Components in compset are: ['datm', 'slnd', 'dice', 'docn', 'drof', 'sglc', 'swav', 'sesp', 'cpl', 'dart']
Traceback (most recent call last):
  File "./create_newcase", line 204, in <module>
    _main_func(__doc__)
  File "./create_newcase", line 187, in _main_func
    run_unsupported=run_unsupported, answer=answer)
  File "/glade/p/work/sacks/cime/scripts/Tools/../../scripts/lib/CIME/case.py", line 745, in configure
    if grid_name in science_support:
TypeError: argument of type 'bool' is not iterable
```

### PR DESCRIPTION
**Pull request by jedwards4b**
Fixes an issue with --user-compset flag, modifies test to include this
Test suite: scripts_regression_tests.py user-compset tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit
Fixes #1315 

User interface changes?: 

Code review:

### CODE REVIEW

File: scripts/lib/CIME/utils.py [hunk 1]
```diff
@@ -26,7 +26,6 @@ def expect(condition, error_msg, exc_type=SystemExit, error_prefix="ERROR:"):
         if logger.isEnabledFor(logging.DEBUG):
             import pdb
             pdb.set_trace()
-        logger.error("%s %s", error_prefix,error_msg, exc_info=True)
```
Comment (billsacks) [inline scripts/lib/CIME/utils.py]: Is this removal intentional? I assume so, but just wanted to double-check since it seems unrelated to the rest of the PR

File: scripts/tests/scripts_regression_tests.py [hunk 2]
```diff
@@ -248,8 +248,11 @@ def test_a_createnewcase(self):
             shutil.rmtree(testdir)
 
         cls._testdirs.append(testdir)
-
-        run_cmd_assert_result(self, "%s/create_newcase --case %s --compset X --res f19_g16 --output-root %s" % (SCRIPT_DIR, testdir, cls._testroot), from_dir=SCRIPT_DIR)
+        if CIME.utils.get_model() == "cesm":
+            pesfile = os.path.join("..","src","drivers","mct","cime_config","config_pes.xml")
+            run_cmd_assert_result(self, "%s/create_newcase --case %s --compset 2000_SATM_XLND_SICE_SOCN_XROF_XGLC_SWAV --user-compset --run-unsupported --pesfile %s --res f19_g16 --output-root %s" % (SCRIPT_DIR, testdir, pesfile, cls._testroot), from_dir=SCRIPT_DIR)
+        else:
+            run_cmd_assert_result(self, "%s/create_newcase --case %s --compset X --res f19_g16 --output-root %s" % (SCRIPT_DIR, testdir, cls._testroot), from_dir=SCRIPT_DIR)
```
Comment (billsacks) [inline scripts/tests/scripts_regression_tests.py]: Thanks for adding a test of user compset! It seems like we still want to do the test without `--user-compset` for cesm too, right? i.e., take that second create_newcase command out of the else clause and just always do it?

File: scripts/lib/CIME/utils.py (hunk 1 above)
Comment (jedwards4b) [inline scripts/lib/CIME/utils.py]: Yes, it was added in a previous pr today, but I didn't like the default behavior.

**Comment by jedwards4b** [comment]
@jgfouca can you run scripts_regression_tests.py on an acme system and make sure that the
new J_TestCreateNewcase works correctly?

**Comment by jgfouca** [comment]
@jedwards4b sure

**Comment by jgfouca** [comment]
@jedwards4b it worked.

**Comment by jedwards4b** [comment]
@billsacks I've added the user-compset test as a separate test and made sure it works in both cesm and acme - good to go?

**Comment by billsacks** [comment]
Yes, looks good - thank you!
